pkg/aks/client.go

```diff
@@ -133,5 +141,45 @@ func GetSecrets(secretsCache wranglerv1.SecretCache, spec *aksv1.AKSClusterConfi
 	cred.AuthBaseURL = spec.AuthBaseURL
 	cred.BaseURL = spec.BaseURL
 
+	if cred.TenantID == "" {
+		cred.TenantID, err = getCachedTenantID(secretClient, cred.SubscriptionID, secret)
+		if err != nil {
+			logrus.Debugf("encountered error while updating secret, ignoring: %v", err)
+		}
+	}
+
 	return &cred, nil
 }
+
+func getCachedTenantID(secretClient wranglerv1.SecretClient, subscriptionID string, secret *v1.Secret) (string, error) {
+	annotations := secret.GetAnnotations()
+	tenantAnno, timestamp := annotations[tenantIDAnnotation], annotations[tenantIDTimestampAnnotation]
+	if tenantAnno != "" && timestamp != "" {
+		parsedTime, err := time.Parse(time.RFC3339, timestamp)
+		if err != nil {
+			return "", err
+		}
+		if parsedTime.Add(tenantIDTimeout).After(time.Now().UTC()) {
+			logrus.Infof("using cached tenant ID")
+			return tenantAnno, nil
+		}
+	}
+	logrus.Infof("looking up tenant ID")
```

Comment:
Should make these log messages more specific, include secret name, but not include any other cred info.

Reply:
These were debug logs that I accidentally left in